rocketpy/utilities.py

```diff
@@ -685,3 +690,71 @@ def get_instance_attributes(instance):
         if not inspect.ismethod(member[1]) and not member[0].startswith("__"):
             attributes_dict[member[0]] = member[1]
     return attributes_dict
+
+
+def save_to_rpy(flight: Flight, filename: str, include_outputs = False):
+    """Saves a .rpy file into the given path, containing key simulation 
+    informations to reproduce the results.
+
+    Parameters
+    ----------
+    flight : rocketpy.Flight
+        Flight object containing the rocket's flight data
+    filename : str
+        Path where the file will be saved in
+    include_output : bool, optional
+        If True, the function will include extra outputs into the file,
+        by default False
+        
+    Returns
+    -------
+    None
+    """
+    file = os.path.splitext(filename)[0] + ".rpy"
+    with open(file, "w") as f:
+        data = {"date": str(date.today()), "version": version("rocketpy")}
+        print(flight)
+        data["simulation"] = flight
+        print(data)
+        print(flight.rocket)
+        json.dump(
+            data,
+            f,
+            cls=RocketPyEncoder,
+            indent=2,
+            include_outputs=include_outputs,
+        )
+
+
+def load_from_rpy(filename: str, resimulate = False):
+    """Loads the saved data from the .rpy into a Flight object.
+
+    Parameters
+    ----------
+    filename : str
+        Path where the file to be loaded is
+    resimulate : bool, optional
+        If True, the function will resimulate the object Flight,
+        by default False
+        
+    Returns
+    -------
+    rocketpy.Flight
+        Flight object containing simulation information from the .rpy file
+    """
+    ext = os.path.splitext(os.path.basename(filename))[1]
+    if ext == ".rpy":
+        with open(filename, "r") as f:
+            data = json.load(f)
+            if data["version"] > version("rocketpy"):
+                warnings.warn("The file was saved in an updated version of",
+                              f"RocketPy (v{data["version"]}), the current",
+                              f"imported module is v{version('rocketpy')}")
+            simulation = json.dumps(data["simulation"])
+            flight = json.loads(
+                simulation,
+                cls=RocketPyDecoder,
+                resimulate=resimulate)
+        return flight
+    else:
+        raise ValueError(f"Invalid file extension: {ext}. Allowed: .rpy")
```

Comment:
You could invert the order of testing here: first you raise the exception, then you run your code.

```suggestion
    if ext != ".rpy":
        raise ValueError(f"Invalid file extension: {ext}. Allowed: .rpy")

    with open(filename, "r") as f:
        data = json.load(f)
        if data["version"] > version("rocketpy"):
            warnings.warn("The file was saved in an updated version of",
                        f"RocketPy (v{data["version"]}), the current",
                        f"imported module is v{version('rocketpy')}")
        simulation = json.dumps(data["simulation"])
        flight = json.loads(
            simulation,
            cls=RocketPyDecoder,
            resimulate=resimulate)
    return flight
```
This way you don't have to ident all our code.